pkg/utils/timeParser.go

```diff
@@ -2,81 +2,45 @@ package utils
-			for i, v := range q1 {
-				if i == 0 {
-					val += v[len(kind)+2:]
-				} else {
-					val += " " + v
-				}
-				if strings.HasSuffix(v, "\"") {
-					break
-				}
-			}
-			params.SearchQuery = strings.ReplaceAll(params.SearchQuery, val, "")
-			params.SearchQuery = strings.ReplaceAll(params.SearchQuery, "earliest=\"", "")
-		}
-		return strings.TrimSuffix(val, "\"")
+// check if the search string contains the earliest or latest time and return the time and the query
+func getQueryTime(kind string, searchQuery string, defaultTime string) (string, string) {
+	if !strings.Contains(searchQuery, kind) {
+		return defaultTime, searchQuery
 	}
+	startIndex := strings.Index(searchQuery, kind)
+	q1 := strings.Fields(searchQuery[startIndex:])
 
-	return defaultTime
-}
-
-// check if the search string contains the earliest or latest time and return the value
-func getAlertTime(kind string, search string, params *splunk.AlertParams, defaultTime string) string {
-	if strings.Contains(search, kind) {
-		startIndex := strings.Index(search, kind)
-		q1 := strings.Fields(search[startIndex:])
+	timeValue := ""
+	if !strings.HasPrefix(q1[0][len(kind)+1:], "\"") {
+		timeValue = q1[0][len(kind)+1:]
+		searchQuery = strings.ReplaceAll(searchQuery, q1[0], "")
 
-		val := ""
-		if !strings.HasPrefix(q1[0][len(kind)+1:], "\"") {
-			val = q1[0][len(kind)+1:]
-			params.SearchQuery = strings.ReplaceAll(params.SearchQuery, q1[0], "")
-		} else {
-			for i, v := range q1 {
-				if i == 0 {
-					val += v[len(kind)+2:]
-				} else {
-					val += " " + v
-				}
-				if strings.HasSuffix(v, "\"") {
-					break
-				}
-			}
-			params.SearchQuery = strings.ReplaceAll(params.SearchQuery, val, "")
-			params.SearchQuery = strings.ReplaceAll(params.SearchQuery, "earliest=\"", "")
+		return strings.TrimSuffix(timeValue, "\""), searchQuery
+	}
+	for i, v := range q1 {
+		switch i {
+		case 0:
+			timeValue += v[len(kind)+2:]
+		default:
+			timeValue += " " + v
+		}
+		if strings.HasSuffix(v, "\"") {
+			break
 		}
-		return strings.TrimSuffix(val, "\"")
 	}
+	searchQuery = strings.ReplaceAll(searchQuery, timeValue, "")
+	searchQuery = strings.ReplaceAll(searchQuery, kind+"=\"", "")
 
-	return defaultTime
+	return strings.TrimSuffix(timeValue, "\""), searchQuery
 }
 
-// get the earliest and latest time from the splunk search is set
-func RetrieveSearchTimeRange(params *splunk.RequestParams) {
-	search := params.SearchQuery
-
-	params.EarliestTime = getSearchTime("earliest", search, params, params.EarliestTime)
-	params.LatestTime = getSearchTime("latest", search, params, params.LatestTime)
-}
+// get the earliest, latest time from the splunk search and also update the search query
+func RetrieveQueryTimeRange(earliestTime string, latestTime string, searchQuery string) (string, string, string) {
 
-func RetrieveAlertTimeRange(params *splunk.AlertParams) {
-	search := params.SearchQuery
+	earliestTime, searchQuery = getQueryTime("earliest", searchQuery, earliestTime)
```

Comment:
This is the only point I want to highlight here, I'm noticing it now but I see it was the case also in the previous version. Is it a problem that `searchQuery` is overridden on line 43?